### PR TITLE
Exporting Context from Native to Flutter

### PR DESCRIPTION
**Pull request by desusai7**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: ios/Classes/RudderSdkFlutterPlugin.m [hunk 1]
```diff
@@ -123,6 +114,10 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
             [RSClient setAnonymousId:[call.arguments objectForKey:@"anonymousId"]];
         }
     }
+    else if ([@"getRudderContext" isEqualToString:call.method]) {
```
Comment (arnab-p) [inline ios/Classes/RudderSdkFlutterPlugin.m:117]:
```suggestion
    } else if ([@"getRudderContext" isEqualToString:call.method]) {
```

File: ios/Classes/RudderSdkFlutterPlugin.m [hunk 2]
```diff
@@ -123,6 +114,10 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
...
+        if ([RSClient sharedInstance] == nil) return;
```
Comment (arnab-p) [inline ios/Classes/RudderSdkFlutterPlugin.m:118]:
```suggestion
        if ([RSClient sharedInstance] == nil) {
          return;
        }
```

File: ios/Classes/RudderSdkFlutterPlugin.m [hunk 3]
```diff
@@ -200,18 +195,18 @@ -(RSOption*) getRudderOptionsObject:(NSDictionary *) optionsDict {
...
     {
-      NSDictionary *integrationsDict = [optionsDict objectForKey:@"integrations"];
-      for(NSString* key in integrationsDict)
-      {
-          [options putIntegration:key isEnabled:[[integrationsDict objectForKey:key] boolValue]];
-      }
+        NSDictionary *integrationsDict = [optionsDict objectForKey:@"integrations"];
+        for(NSString* key in integrationsDict)
+        {
+            [options putIntegration:key isEnabled:[[integrationsDict objectForKey:key] isEqual: @1] ? YES : NO];
```
Comment (arnab-p) [inline ios/Classes/RudderSdkFlutterPlugin.m:208]: Can you please explain why this would be necessary?

**Review by arnab-p — CHANGES_REQUESTED** [review]

File: ios/Classes/RudderSdkFlutterPlugin.m (hunk 3 above)
Comment (desusai7) [reply ios/Classes/RudderSdkFlutterPlugin.m:208]: isEnabled parameter is Expecting a type of`BOOL` and if we try to send the value by casting using `boolValue` as before it is considering the value as true if it is not null. So even if it's false it is being considered true. So we are directly sending a `BOOL` to overcome this behaviour

**Review by arnab-p — APPROVED** [review]

**Pull request merged**